bofire/benchmarks/aspen_benchmark.py

```diff
@@ -0,0 +1,199 @@
+        """
+        if os.path.exists(filename):
+            self.filename = filename
+        else:
+            raise ValueError("Unable to find Aspen file " + filename)
+
+        self.translate_into_aspen_readable = translate_into_aspen_readable
+        self._domain = domain
+        # Get the variable names (keys) from the domain to access them later easily.
+        self.keys = [self.domain.inputs.get_keys(), self.domain.outputs.get_keys()]
+        self.input_keys = self.domain.inputs.get_keys()
+        self.output_keys = self.domain.outputs.get_keys()
+        self.additional_outputs = []
+        if additional_outputs is not None:
+            self.additional_outputs = additional_outputs
+
+        for key_list in [self.input_keys, self.output_keys]:
+            for key in key_list:
+                # Check, if every input and output variable has a path to Aspen provided.
+                if key not in paths.keys():
+                    raise ValueError("Path for " + key + " is not provided.")
+
+        self.paths = paths
+        self.aspen_is_running = False
+
+    # Start Aspen
+    def start_aspen(self):
+        """Starts Aspen plus and opens desired simulation file.
+
+        Raises:
+            ValueError: In case it is not possible to start Aspen plus.
+        """
+        import win32com.client as win32
+
+        logger.info("Starting Aspen plus")
+        # Aspen should be accessible from every function as a global variable.
+        global aspen
+        try:
+            aspen = win32.Dispatch("Apwn.Document")
+            aspen.InitFromFile2(os.path.abspath(self.filename))
+            self.aspen_is_running = True
+        except OSError as e:
+            logger.exception(e)
+            raise ValueError(e)
+
+    # Run simulation in Aspen
+    def _f(self, candidates: pd.DataFrame) -> pd.DataFrame:
+        """Function evaluation of the Aspen plus benchmark. Passes input values to Aspen, runs
+        a simulation and reads the output values from Aspen. Checks for errors/warnings from Aspen.
+
+        Args:
+            candidates (pd.DataFrame): Input values that need to be passed to Aspen.
+            Function can take either one input vector as a row in the input dataframe or multiple input vectors.
+
+        Returns:
+            pd.DataFrame: Output values from Aspen. The dataframe includes valid_(variable_name) columns for
+            each output variable when the simulation went successful.
+        """
+
+        # Only start Aspen, when it is not already blocking.
+        if self.aspen_is_running is False:
+            self.start_aspen()
+
+        # Make inputs Aspen-readable
+        if self.translate_into_aspen_readable is not None:
+            candidates = self.translate_into_aspen_readable(
+                domain=self.domain,  # type: ignore
+                candidates=candidates,
+            )
+
+        y_outputs = {}
+        add_outputs = {}
```

Comment:
```suggestion
        add_outputs = {key: [] for key in self.additional_outputs}
```

And then delete the following 5 rows.